### DISCUSSION TITLE
The SQL execution generated by the export Oracle table data function will report an error

### DISCUSSION THREAD

**Discussion by bianyun1981** [dbeaver/dbeaver#15929] [Ideas]
#### System information: 
- Operating system (distribution) and version: Windows 10 Enterprise Edition (Simplified Chinese), 21H1
- DBeaver version: 21.3.4.202202051448

#### Connection specification:
- Database name and version: Oracle 11.2.0.1.0
- Driver name: Oracle (oracle.jdbc.OracleDriver)

#### Describe the problem you're observing:
When I used DBeaver to export Oracle table data, I found that the generated SQL statement used the feature of inserting multiple rows in a single statement that Oracle does not support. An example of the generated SQL statement is as follows:
```
INSERT INTO ISC_USR_PRIPWD (ID,NAME,LOGIN_NAME,PASSWORD) VALUES
	 ('40286c007aebcb8d017aebf4d4fe0004','zjb11','zjb11','C769CF2B9C97D9AD211D60D6ADAB962F'),
	 ('40286c007aefed1f017af5505e2e0044','zjb730','zjb730','7F7E23A043F0090AE8603472CE061B6D'),
	 ('8a9680817cc6286f017cc64182630004','fsir_002','fsir_003','7F7E23A043F0090AE8603472CE061B6D');
```

**Comment by uslss** [comment]
@bianyun1981 
the number of the data rows per statement could be changed in Data Export preferences
![image](https://user-images.githubusercontent.com/31996417/153228163-3817b855-cccb-48aa-bd00-ec4321dcdb40.png)

**Comment by bianyun1981** [comment]
@uslss 
Thanks for pointing out the location of this config. After testing, it is found that as long as it is set to 1, the exported SQL format returns to normal.

Although modifying the value of this configuration can temporarily solve this problem, there seems to be a little inconvenient to use, that is: if I am used to setting this to a value greater than 1 under the MySQL connection (ie: a single statement inserts multiple rows ), then when I switch back and forth between MySQL and Oracle, when exporting data, I have to change the value of this configuration every time.

I feel that whether this configuration takes effect should be pre-judged with the type of connection. For example, Oracle does not support the syntax of inserting multiple rows of data in a single statement, so the value of this configuration should be fixed at 1, and it is not allowed to be modified.

**Comment by LonwoLonwo** [comment]
Hello @bianyun1981 

We have a special setting for the multiple exports for the Oracle database:

![2022-02-09 20_15_51-Data Transfer](https://user-images.githubusercontent.com/45152336/153254128-1717f5b0-48cb-4494-99b4-5369a1c1dfda.png)

It will return something like this:

```sql
INSERT ALL
	INTO public.newtable (column1,column2,column3) VALUES ('dogThe','dogThe quick brown fox jumps over the lazy',300393)
	INTO public.newtable (column1,column2,column3) VALUES ('the la','bro',129497)
	INTO public.newtable (column1,column2,column3) VALUES ('fox jumps over th','the lazy dogThe quick brown f',332652)
SELECT 1 FROM DUAL;
```

Thanks for the idea, but DBeaver usually tries to save your preferences/settings. If we will change export settings depending on the Database - it can be confusing.

**Comment by bianyun1981** [comment]
@LonwoLonwo 

Thank you for your reply. I didn't know that Oracle supports this form of syntax for inserting multiple rows with a single statement, so this configuration (Data rows per statement) makes sense for Oracle.

As for the fact that DBeaver saves preferences for different export settings based on different database types can be confusing, I agree that it is.

However, if a configuration (such as a configuration in export table data: "Upsert keyword") needs to be set to different values for different types of databases, then the saving of preferences for this configuration should not be saved globally. Even if this configuration value is not saved by database type, it should at least save one configuration value per connection, so that each connection only needs to set this configuration once. Otherwise, when switching between different types of databases, you may need to modify this configuration every time you switch.

**Reply by uslss** [reply]
Looks interesting. Feel free to create a feature request.
BTW, you could try to use tasks https://github.com/dbeaver/dbeaver/wiki/Task-Management

**Reply by bianyun1981** [reply]
Thank you for your reply, I haven't used tasks before, but it seems to work well, I will use it later.

By the way, thank you and your team for developing such a great open source software DBeaver, it's powerful and much better than a lot of paid software.
